### PR TITLE
config: make auto id step can be configured.

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
Getting auto increment id in TiDB need start a new transaction. Some application run out of auto increment id quickly and request id transaction make insert operation slowly.

### What is changed and how it works?
add `auto-id-step` to config, user can set up this value to improve performance.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -192,6 +192,7 @@ type Performance struct {
 	PseudoEstimateRatio float64 `toml:"pseudo-estimate-ratio" json:"pseudo-estimate-ratio"`
 	ForcePriority       string  `toml:"force-priority" json:"force-priority"`
 	BindInfoLease       string  `toml:"bind-info-lease" json:"bind-info-lease"`
+	AutoIDStep          int64   `toml:"auto-id-step" json:"auto-id-step"`
```
Comment (coocood) [inline config/config.go]: `uint64` is better

**Comment by coocood** [comment]
We should validate the config value, make sure it greater than 0

**Comment by ngaut** [comment]
I think the better way is to make the default ID step big enough, instead of adding a new configuration.

**Comment by coocood** [comment]
@ngaut 
Make the step too big makes the user feel bad, worries about running out of ID soon.
Some users have complained about the default step is too big already.

**Comment by coocood** [comment]
@imtbkcat 
We should return an error when the config value is invalid.

**Comment by imtbkcat** [comment]
@coocood added.

**Comment by shenli** [comment]
@coocood Could we increase the default value a bit (100k for example)?

**Comment by coocood** [comment]
Increase the default value has the problem make the autoid growing rapidly, it's more likely to run out of id.

File: tidb-server/main.go [hunk 2]
```diff
@@ -492,6 +493,8 @@ func setGlobalVars() {
 
 	tikv.CommitMaxBackoff = int(parseDuration(cfg.TiKVClient.CommitTimeout).Seconds() * 1000)
 	tikv.PessimisticLockTTL = uint64(parseDuration(cfg.PessimisticTxn.TTL).Seconds() * 1000)
+
+	autoid.SetStep(cfg.Performance.AutoIDStep)
```
Comment (crazycs520) [inline tidb-server/main.go]: Please add test for this.

File: config/config.toml.example [hunk 3]
```diff
@@ -174,6 +174,9 @@ force-priority = "NO_PRIORITY"
 # Bind info lease duration, which influences the duration of loading bind info and handling invalid bind.
 bind-info-lease = "3s"
 
+# The number of auto increment id cached in memory.
```
Comment (winkyao) [inline config/config.toml.example]: Please add a test in config_test.go

File: config/config.go [hunk 4]
```diff
@@ -564,6 +566,9 @@ func (c *Config) Valid() error {
 	if c.TiKVClient.MaxTxnTimeUse == 0 {
 		return fmt.Errorf("max-txn-time-use should be greater than 0")
 	}
+	if c.Performance.AutoIDStep <= 0 {
```
Comment (winkyao) [inline config/config.go]: I think we should prevent the `step` is too big, what about setting 50w as the max `step`?

File: tidb-server/main.go (hunk 2 above)
Comment (bb7133) [inline tidb-server/main.go]: Please remove the comment:

https://github.com/pingcap/tidb/blob/cf5f42b8ece0fad2b3f93c6753747b392f33f4c8/meta/autoid/autoid.go#L70

**Comment by coocood** [comment]
@imtbkcat 
We have got a better solution than adding a config option.
By adding a Time field in ID allocator, we can calculate the next step with a simple function
```go
func NextStep(curStep int, consumeDur time.Duration) int
```

**Pull request closed without merging**